elasticapm/base.py

```diff
@@ -559,6 +565,13 @@ def _get_stack_info_for_trace(
             locals_processor_func=locals_processor_func,
         )
 
+    def _excepthook(self):
+        exec_info = sys.exc_info()
+        if self.was_default_excepthook_overridden_by_user:
+            self.original_excepthook(*exec_info)
```

Comment:
Could it make sense to call this in a `try/finally` block to ensure that our `capture_exception` method is called even if the original exception hook raises an exception for some reason?

Reply:
Seems reasonable to me. 👍

Reply:
That makes completely sense. Good catch :+1:

Reply:
I was thinking about this modification and ended up with a question. What If the exception hook defined by the user raises an exception, what should be done in such case? Shall the code capture it and handle using `self.capture_exception` or shall let it bubble up?

Example:
```
import sys

sys.excepthook = lambda x, y, z: 1/0

raise Exception()
```

This piece of code results in:
```
>>> import sys
>>> 
>>> sys.excepthook = lambda x, y, z: 1/0
>>> 
>>> 
>>> raise Exception()
Error in sys.excepthook:
Traceback (most recent call last):
  File "<stdin>", line 1, in <lambda>
ZeroDivisionError: division by zero

Original exception was:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception
```

So, in this case it is also possible to catch this exception and call elastic handler, similar to the following snippet 
```
    def _excepthook(self):
        exec_info = sys.exc_info()
        try:
            self.original_excepthook(*exec_info)
        except Exception:
            self.capture_exception(sys.exc_info(), handled=False)
        finally:
            self.capture_exception(exec_info, handled=False)
```

Any thoughts here?

Reply:
Hmm. I guess we should probably collect both? Seems unlikely to be an edge case we ever see (I don't think the average user is replacing excepthook) but probably worth covering?

Reply:
@beniwohli do you have thoughts here?

Reply:
IMO the current implementation is fine, as it will capture both exceptions. But I agree that it is probably an edge case.